clang/lib/CIR/CodeGen/CIRGenExprComplex.cpp

```diff
@@ -164,14 +173,106 @@ LValue ComplexExprEmitter::emitBinAssignLValue(const BinaryOperator *e,
 mlir::Value ComplexExprEmitter::emitCast(CastKind ck, Expr *op,
                                          QualType destTy) {
   switch (ck) {
+  case CK_Dependent:
+    llvm_unreachable("dependent cast kind in IR gen!");
```

Comment:
Can we improve this message, so that it is clear why this should never come here.

Reply:
We can change it to something like "dependent type must be resolved before the Codegen" 🤔 @xlauko

Reply:
maybe "dependent type must be resolved before the CIR codegen"?